### PR TITLE
feat: support chrome headless mode

### PR DESCRIPTION
**Pull request by OrKoN**
Issues: #8173

### CODE REVIEW

File: docs/api.md [hunk 1]
```diff
@@ -580,7 +580,7 @@ This methods attaches Puppeteer to an existing browser instance.
 #### puppeteer.defaultArgs([options])
 
 - `options` <[Object]> Set of configurable options to set on the browser. Can have the following fields:
-  - `headless` <[boolean]> Whether to run browser in [headless mode](https://developers.google.com/web/updates/2017/04/headless-chrome). Defaults to `true` unless the `devtools` option is `true`.
+  - `headless` <[boolean]|"chrome"> Whether to run browser in [headless mode](https://developers.google.com/web/updates/2017/04/headless-chrome). Defaults to `true` unless the `devtools` option is `true`. "chrome" is a new experimental headless mode (you at your own risk).
```
Comment (ergunsh) [inline docs/api.md]: This is typo I think :)

```suggestion
  - `headless` <[boolean]|"chrome"> Whether to run browser in [headless mode](https://developers.google.com/web/updates/2017/04/headless-chrome). Defaults to `true` unless the `devtools` option is `true`. "chrome" is a new experimental headless mode (use at your own risk).
```

File: docs/api.md [hunk 2]
```diff
@@ -646,7 +646,7 @@ try {
 - `options` <[Object]> Set of configurable options to set on the browser. Can have the following fields:
   - `product` <[string]> Which browser to launch. At this time, this is either `chrome` or `firefox`. See also `PUPPETEER_PRODUCT`.
   - `ignoreHTTPSErrors` <[boolean]> Whether to ignore HTTPS errors during navigation. Defaults to `false`.
-  - `headless` <[boolean]> Whether to run browser in [headless mode](https://developers.google.com/web/updates/2017/04/headless-chrome). Defaults to `true` unless the `devtools` option is `true`.
+  - `headless` <[boolean]|"chrome"> Whether to run browser in [headless mode](https://developers.google.com/web/updates/2017/04/headless-chrome). Defaults to `true` unless the `devtools` option is `true`. "chrome" is a new experimental headless mode (you at your own risk).
```
Comment (ergunsh) [inline docs/api.md]:
```suggestion
  - `headless` <[boolean]|"chrome"> Whether to run browser in [headless mode](https://developers.google.com/web/updates/2017/04/headless-chrome). Defaults to `true` unless the `devtools` option is `true`. "chrome" is a new experimental headless mode (use at your own risk).
```

Comment (ergunsh) [inline docs/api.md]: I actually wonder what is the difference between `chrome` headless and `true` headless. Is there a documentation on the difference? 🤔

If yes, can we link it in here?
If no, can we add a bit more explanation on the difference between these two modes to this documentation?

**Review by ergunsh — APPROVED** [review]
LGTM with comments

File: docs/api.md (hunk 2 above)
Comment (OrKoN) [reply docs/api.md]: I think the new mode is experimental for now so we don't want to advertise it too much. We can document it once it's more stable. Let's chat about the difference offline! For now we can just monitor and try to address CI failures to make it ready to be used.

Comment (ergunsh) [reply docs/api.md]: Makes sense, SGTM! 🥳

**Pull request merged**